### ISSUE TITLE
[BUG] Compilation error: field 'transport' will be initialized after field 'version' [-Werror,-Wreorder-ctor]

### ISSUE THREAD

**Issue by mateuszzz88** [pistacheio/pistache#928]
## Current Behavior
Initializers are in wrong order, this causes error with strict compiler flags
`src/common/http.cc:1147:15: error: field 'transport' will be initialized after field 'version' [-Werror,-Wreorder-ctor]`

## Expected behavior
They should be in correct order.

## Possible Solution
Reorder initialization list to match member list.

While at it consider moving toward "Rule of 0". Not possible 100%, but `armed` and `timerFd` could have been initialized in class definition and omitted in initialization list.

## Steps to reproduce the bug
Compile project with `-Werror` compiler flag. 

## Context (Environment)

- Pistache version: commit eb57620
- Operating System: ubuntu
- Architecture: amd64
- Installed from: source
- Used `pkg-config`: no

compiler: clang++-10

Screenshots, logs, stacktraces or other relevant information here.
```
FAILED: separate_libs/3rdparty/pistache/pistache-6.6.6/src/CMakeFiles/pistache.dir/common/http.cc.o 
/usr/bin/clang++-10 -DBOOST_FILESYSTEM_VERSION=3 -DBOOST_NO_AUTO_PTR -DBOOST_NO_CXX11_EXPLICIT_CONVERSION_OPERATORS=1 -DBORG_MAKETUI -DBORG_USE_BDB -DBORG_USE_GMP -DBORG_USE_UPSCALEDB -DBUILD_ALPHA -DBUILD_PARTNER_LIVE_POLAND -DENABLE_EQUILIBRIUM -DENABLE_EQ_SERVICE_WS -DENABLE_GUS_WS -DENABLE_KYCGATEWAY -DENABLE_KYCORACLE -DENABLE_KYC_GW_WS -DENABLE_KYC_ORACLE_WS -DENABLE_NBP -DENABLE_OBIM -DENABLE_PUBLISHER -DENABLE_PUB_WS -DENABLE_SUPERVISOR -DENABLE_SUPERVISOR_WS -DLINUX64 -DMAKEMEGOD -DONLY_C_LOCALE=1 -DWITH_OPENSSL -D_FORTIFY_SOURCE=2 -D_REENTRANT -Dbassert=assert -I/home/mateuszl/src/borg/separate_libs/libs/serialization/include -I_deps/rapidjson-src/include -I/home/mateuszl/src/borg/separate_libs/3rdparty/pistache/pistache-6.6.6/src/../include -isystem /usr/local/linux64-hard-2.17_2020-07/include -O2 -g -DNDEBUG -DNDEBUG -Wall -Werror=return-type -Werror=array-bounds -Werror=reorder -Og -gline-tables-only -fPIC -ftemplate-backtrace-limit=0   -Wall -Wno-sign-compare -Wno-unknown-pragmas -Wformat -Wformat-security -Werror=format-security -Wno-mismatched-tags -fdiagnostics-color=always -fvisibility=hidden  -ftemplate-depth=1024 -Werror=uninitialized -Wno-inconsistent-missing-override -fvisibility-inlines-hidden -fPIC   -pthread -Wall -Wextra -Wpedantic -Wconversion -Wno-sign-conversion -Wno-missing-field-initializers -std=c++17 -MD -MT separate_libs/3rdparty/pistache/pistache-6.6.6/src/CMakeFiles/pistache.dir/common/http.cc.o -MF separate_libs/3rdparty/pistache/pistache-6.6.6/src/CMakeFiles/pistache.dir/common/http.cc.o.d -o separate_libs/3rdparty/pistache/pistache-6.6.6/src/CMakeFiles/pistache.dir/common/http.cc.o -c /home/mateuszl/src/borg/separate_libs/3rdparty/pistache/pistache-6.6.6/src/common/http.cc
/home/mateuszl/src/borg/separate_libs/3rdparty/pistache/pistache-6.6.6/src/common/http.cc:1147:15: error: field 'transport' will be initialized after field 'version' [-Werror,-Wreorder-ctor]
            , transport(transport_)
              ^
1 error generated.
```

**Comment by Tachi107** [comment]
This will be fixed by #897.
I would recommend you to stop compiling third party projects with `-Werror` anyway.

**Comment by mateuszzz88** [comment]
> This will be fixed by #897.
Nice
> I would recommend you to stop compiling third party projects with `-Werror` anyway.
Someday, certainly :-) Right now it would require cleaning up the oldest and most unfamiliar parts of our cmake...
